common/actions.js

```diff
@@ -30,6 +30,21 @@ export const addUserSays = () => {
   }
 }
 
+export const addResponse = () => {
+  return {
+    type: "ADD_RESPONSE",
+    id: intentId++
+  }
+}
+
+export const addConversationComponent = (cType) => {
+  return {
+    type: "ADD_CONVERSATION_COMPONENT",
+    id: (intentId+=10)-10,
+    cType: cType,
```

Comment:
(Just a comment or tip) "cType" and "cType: cType" will have the same effect, which means you can get cType when you call "action.cType" in reducer no matter which statement you used here.